### ISSUE TITLE
Sidecar injector TLS handshake error

### ISSUE THREAD

**Issue by veselfil** [dapr/dapr#2783]
## In what area(s)?

/area runtime

## What version of Dapr?

1.0.0-rc1

## Expected Behavior

The DAPR runtime should refresh any TLS certificates it needs to function.

## Actual Behavior

DAPR Sidecar Injector stopped working (stopped injecting sidecars, 0 restarts on the pod) after ~9 days of running. All 3 replicas failed simultaneously. We have mTLS disabled on the deployment.

Restarting the dapr-sidecar-injector deployment has resolved the issue, but presumably only temporarily.

## Steps to Reproduce the Problem

I unfortunately do not have any reproduction steps other than install DAPR and wait. The last time that this has happened our deployment has been running for roughly 8-9 days.

Here's our HelmRelease file with the configuration we use.

```yaml
apiVersion: helm.toolkit.fluxcd.io/v2beta1
kind: HelmRelease
metadata:
  name: dapr
  namespace: dapr-system
spec:
  releaseName: dapr
  interval: 5m
  chart:
    # Chart: https://github.com/dapr/dapr/tree/master/charts/dapr
    spec:
      sourceRef:
        kind: HelmRepository
        name: acr
        namespace: gotk-system
      chart: dapr
      version: 1.0.0-rc.1
      interval: 1m
  upgrade:
    remediation:
      remediateLastFailure: true
  values:
    global:
      registry: # our container registry
      logAsJson: true
      mtls:
        enabled: false
      ha:
        enabled: true
    dapr_placement:
      logLevel: 'debug'
    dapr_sidecar_injector:
      logLevel: 'debug'
    dapr_operator:
      logLevel: 'debug'
    dapr_dashboard:
      replicaCount: 1
      logLevel: info
      image:
        registry: # our container registry
        image: dashboard
        tag: '0.3.0'
      ports:
        protocol: TCP
        port: 80
        targetPort: 8080
```

## Release Note

RELEASE NOTE: **FIX**: TLS handshake bug

**Comment by luk355** [comment]
This issue is getting more frequent now, happening roughly once per day. We didn't find a reason why it started to appear more often.

What we don't fully understand is why we see the TSL error in the first place when having the mTSL turned off - see the Helm values file above.

**Comment by skordesign** [comment]
I have same problem when sidecar-injector inject daprd container to pod, but it only happens in Azure Kubernetes Services.
Working normally in minikube. Any advice?

**Comment by veselfil** [comment]
Looks like we have figured out what our issue was. 

We were using Helm Controller from GOTK to deploy DAPR to our AKS cluster. For some reason, something was triggering Helm upgrade for the deployment. When the deployment got upgraded, nothing happened since the version stayed the same, except for the `dapr-sidecar-injector-cert` secret which got generated again, together with the `MutatingWebhook` manifest for `sidecar-injector`.
The secret however is mounted to the sidecar injector as a file, so the updated certificate is not used there. This results in a state where K8s already has the new `caBundle`, but sidecar injector still has the old certificate, resulting in a error. Restarting sidecar injector fixes the issue, because it loads the new certificate at startup.

This issue is fixed in 1.0.0. For us, we've fixed the issue by modifying the Helm chart for 1.0.0-rc1 and using that instead. See the changes here: https://github.com/veselfil/dapr/commit/07cf0c00361c6ba1232962a27629115f8246ddf9
